cou/cli.py

```diff
@@ -42,110 +71,333 @@ def parse_args(args: Any) -> argparse.Namespace:
-        type=str.upper,
-        help="Set the logging level. Defaults to INFO. This only affects stdout. The logfile "
-        "will be always DEBUG. The file location is in COU_DATA/log. You can set the "
-        "COU_DATA via environment variable. However it needs to be plugged to snap.",
+        "--help",
+        "-h",
+        action="help",
+        default=argparse.SUPPRESS,
+        help="Show this help message and exit.",
     )
-    parser.add_argument(
+
+    # Configure subcommand parser
+    subparsers = parser.add_subparsers(
+        title="Commands",
+        dest="command",
+        help="For more information about a command, run 'cou help <command>'.",
+    )
+
+    # Arg parser for "cou help" sub-command
+    help_parser = subparsers.add_parser(
+        "help",
+        usage="cou help [command]",
+    )
+    help_parser.add_argument(
+        "subcommand",
+        nargs="?",
+        choices=["plan", "run", "all"],
+        default="all",
+        type=str,
+        help="A sub-command to get information of.",
+    )
+
+    # Define common options for subcommands and their child commands
+    subcommand_common_opts_parser = argparse.ArgumentParser(add_help=False)
+    subcommand_common_opts_parser.add_argument(
         "--model",
         default=None,
         dest="model_name",
+        type=str,
         help="Set the model to operate on. If not set it gets the model name in this order:\n"
         "  1 - Environment variable JUJU_MODEL,"
         "  2 - Environment variable MODEL_NAME,"
         "  3 - Current active juju model",
     )
-    parser.add_argument(
+    subcommand_common_opts_parser.add_argument(
+        "--parallel",
+        help="Run upgrade steps in parallel where possible.",
+        default=False,
+        action="store_true",
+    )
+
+    # quiet and verbose options are mutually exclusive
+    group = subcommand_common_opts_parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--verbose",
+        "-v",
+        default=0,
+        action="count",
+        dest="verbosity",
+        help="Increase logging verbosity in STDOUT. Repeat the 'v' in the short option "
+        "for more detail. Maximum verbosity is obtained with 4 (or more) "
+        "v's, i.e. -vvvv. \nNote that this doesn't affect the verbosity in logfile, "
+        "which will always have the maximum verbosity.",
+    )
+    group.add_argument(
+        "--quiet",
+        "-q",
+        default=False,
+        action="store_true",
+        dest="quiet",
+        help="Disable output in STDOUT.",
+    )
+
+    # Define options specific to data-plane child command and make them mutually exclusive
+    dp_subparser = argparse.ArgumentParser(add_help=False)
+    dp_upgrade_group = dp_subparser.add_mutually_exclusive_group()
+    dp_upgrade_group.add_argument(
+        "--machine",
+        "-m",
+        action="append",
+        help="Specify machines ids to upgrade.",
```

Comment:
```suggestion
        help="Specify machine ids to upgrade.",
```

Reply:
Also, we don't clarify how to specify them (e.g. comma separated, or via multiple `-m` options).
I know we aren't actually reading those variables yet, so... should we even include these options? It would mislead users into thinking that data plane upgrades are already implemented.

Reply:
perhaps we could merge this section of the code in a separate branch for now

Reply:
My idea of creating these placeholder options was to construct as much CLI as possible so future developer only needs to read the values without needing to implement extra parsing functionalities. If you are worried about misleading users,  maybe we can let the program exit with "Data plane components upgrade is not yet supported." message when `data-plane` is specified.

Reply:
Yes, given the desire to do a release for the control-plane part I think we should not have the data-plane option at all.
I do agree that starting to design this is useful, hence the idea of merging these lines elsewhere (instead of throwing them away)

Reply:
I'm not against merging it to a separate branch. But if we do, I believe that we should push everything else "data-plane"-related (e.g. PR #54 ) to that branch also so the main branch is clean with just control-plane code.

Reply:
For future reference, based on discussion elsewhere, we agreed to hide upgrade-group options until data-plane upgrade functionality is implemented.